File: frontend/gestionale-bertozzi/src/app/models/TemplatePianiSviluppo/template-piano-sviluppo.ts
import { TemplateAttivita } from './template-attivita';

export class TemplatePianoSviluppo {
    id?: number;
    tipologiaCommessaId: number = 0;
    descrizione: string = '';
    attivita?: TemplateAttivita[];
    ordine: number = 0;


    static map(piano: any): TemplatePianoSviluppo {
        let t = Object.assign(new TemplatePianoSviluppo(), piano) as TemplatePianoSviluppo;

        if (piano.attivita && Array.isArray(piano.attivita)) {
            t.attivita = piano.attivita.map((a: any) => TemplateAttivita.map(a));
        }
        return t;
    }

    static mapArray(arr: any[]): TemplatePianoSviluppo[] {
        console.log("Mapping array of TemplatePianoSviluppo:", arr);
        return (arr || []).map(c => TemplatePianoSviluppo.map(c));
    }
}

File: backend/NemesiLIB/Model/TemplatePianiSviluppo/TemplatePianoSviluppo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NemesiCOMMONS.Models;

namespace NemesiLIB.Model.PianiSviluppo
{
    public class TemplatePianoSviluppo : IAuditable
    {
        public int Id { get; set; }
        public int TipologiaCommessaId { get; set; }
        public string Descrizione { get; set; }
        public int Ordine { get; set; }
        public virtual List<TemplateAttivita>? Attivita { get; set; }
        public DateTime DataCreazione { get; set; }
        public DateTime? DataModifica { get; set; }
        public string? UtenteCreazione { get; set; }
        public string? UtenteModifica { get; set; }
    }
}
